beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go

```diff
@@ -163,39 +161,36 @@ func (vs *Server) getLocalPayload(ctx context.Context, blk interfaces.ReadOnlyBe
 			SuggestedFeeRecipient: val.FeeRecipient[:],
 		})
 		if err != nil {
-			return nil, false, err
+			return nil, err
 		}
 	default:
-		return nil, false, errors.New("unknown beacon state version")
+		return nil, errors.New("unknown beacon state version")
 	}
 	payloadID, _, err := vs.ExecutionEngineCaller.ForkchoiceUpdated(ctx, f, attr)
 	if err != nil {
-		return nil, false, errors.Wrap(err, "could not prepare payload")
+		return nil, errors.Wrap(err, "could not prepare payload")
 	}
 	if payloadID == nil {
-		return nil, false, fmt.Errorf("nil payload with block hash: %#x", parentHash)
+		return nil, fmt.Errorf("nil payload with block hash: %#x", parentHash)
 	}
-	payload, bundle, overrideBuilder, err := vs.ExecutionEngineCaller.GetPayload(ctx, *payloadID, slot)
+	res, err := vs.ExecutionEngineCaller.GetPayload(ctx, *payloadID, slot)
 	if err != nil {
-		return nil, false, err
-	}
-	bundleCache.add(slot, bundle)
-	warnIfFeeRecipientDiffers(payload, val.FeeRecipient)
-	localValueGwei, err := payload.ValueInGwei()
-	if err == nil {
-		log.WithField("value", localValueGwei).Debug("received execution payload from local engine")
+		return nil, err
 	}
-	return payload, overrideBuilder, nil
+
+	warnIfFeeRecipientDiffers(res.ExecutionData.FeeRecipient(), val.FeeRecipient[:])
+	log.WithField("value", res.Bid).Debug("received execution payload from local engine")
+	return res, nil
 }
 
 // warnIfFeeRecipientDiffers logs a warning if the fee recipient in the included payload does not
 // match the requested one.
-func warnIfFeeRecipientDiffers(payload interfaces.ExecutionData, feeRecipient primitives.ExecutionAddress) {
+func warnIfFeeRecipientDiffers(payload, val []byte) {
```

Comment:
these parameters feel harder to read, is there a better name for this? ( not a big deal , just a comment)

Reply:
I thought this was more clear since both parameters are fee recipients and now have the same type (previously one was called `feeRecipient`, which is also in the fuction name, and yeah they're both fee recipients). The function name tells you that you are comparing fee recipients, I figured it would be clear that one is the fee recipient from the payload, and the other is the one from the validator registration. Maybe the `val` name is too ambiguous and something like this would be more clear?
How about `func warnIfFeeRecipientDiffers(fromPayload, fromRegistration []byte) {`

Reply:
> func warnIfFeeRecipientDiffers(payload, registration []byte) {
yeah i think that may be better 👍

Reply:
Even more simple: `func warnIfFeeRecipientDiffers(want, got []byte) {`